### PR TITLE
module/dns: aws now uses the route53 module

### PR DESCRIPTION
**Pull request by kalmog**

### CODE REVIEW

**Comment by kalmog** [comment]
test this please

File: platforms/aws/main.tf [hunk 1]
```diff
@@ -194,3 +187,23 @@ module "workers" {
...
+  etcd_ips                  = "${module.etcd.ip_addresses}"
+  external_etcd             = "${var.tectonic_etcd_servers}"
+  master_count              = "${var.tectonic_master_count}"
+  cluster_name              = "${var.tectonic_cluster_name}"
+  cluster_id                = "${module.tectonic.cluster_id}"
+  base_domain               = "${var.tectonic_base_domain}"
+  tectonic_external_vpc_id  = "${module.vpc.vpc_id}"
+  api_external_elb_dns_name = "${module.vpc.aws_api_external_dns_name}"
+  api_external_elb_zone_id  = "${module.vpc.aws_elb_api_external_zone_id}"
+  api_inernal_elb_dns_name  = "${module.vpc.aws_api_internal_dns_name}"
```
Comment (s-urbaniak) [inline platforms/aws/main.tf]: s/api_inernal_elb_dns_name/api_internal_elb_dns_name

Comment (kalmog) [reply platforms/aws/main.tf]: corrected

**Comment by cpanato** [comment]
retest this please

**Comment by kalmog** [comment]
test this please

**Comment by squat** [comment]
@kalmog can you please rebase?

**Comment by squat** [comment]
Looks like there are still conflicts with master. Also, can we rebase rather than merge?

**Comment by kalmog** [comment]
test this please

**Comment by mxinden** [comment]
We did some changes (https://github.com/coreos/tectonic-installer/pull/2082) to the testing process. Please rebase on to current master, so that the `basic-tests` PR status is reported correctly.

File: modules/aws/vpc/master-elb.tf [hunk 2]
```diff
@@ -31,25 +31,12 @@ resource "aws_elb" "api_internal" {
...
-    evaluate_target_health = true
-  }
-}
-
 resource "aws_elb" "api_external" {
-  count           = "${var.public_endpoints}"
+  count           = "${var.public_master_endpoints}"
   name            = "${var.custom_dns_name == "" ? var.cluster_name : var.custom_dns_name}-ext"
-  subnets         = ["${var.subnet_ids}"]
+  subnets         = ["${split(",", var.external_vpc_id == "" ? join(",", aws_subnet.master_subnet.*.id) :  join(",", data.aws_subnet.external_master.*.id))}"]
```
Comment (s-urbaniak) [inline modules/aws/vpc/master-elb.tf]: nit: Can we deduplicate this by adding a `local`?

File: modules/aws/vpc/master-elb.tf [hunk 3]
```diff
@@ -77,24 +64,11 @@ resource "aws_elb" "api_external" {
...
-    evaluate_target_health = true
-  }
-}
-
 resource "aws_elb" "console" {
   name            = "${var.custom_dns_name == "" ? var.cluster_name : var.custom_dns_name}-con"
-  subnets         = ["${var.subnet_ids}"]
-  internal        = "${var.public_endpoints ? false : true}"
-  security_groups = ["${var.console_sg_ids}"]
+  subnets         = ["${split(",", var.external_vpc_id == "" ? join(",", aws_subnet.master_subnet.*.id) :  join(",", data.aws_subnet.external_master.*.id))}"]
```
Comment (s-urbaniak) [inline modules/aws/vpc/master-elb.tf]: see above

File: platforms/aws/main.tf [hunk 4]
```diff
@@ -194,3 +191,28 @@ module "workers" {
   ign_max_user_watches_id           = "${module.ignition_workers.max_user_watches_id}"
   ign_s3_puller_id                  = "${module.ignition_workers.s3_puller_id}"
 }
+
+module "dns" {
+  source                         = "../../modules/dns/route53"
```
Comment (s-urbaniak) [inline platforms/aws/main.tf]: let's add a newline between `source` and the variables and sort variable names lexically.

**Comment by s-urbaniak** [comment]
@kalmog additionally: can you rename, better yet https://github.com/coreos/tectonic-installer/pull/2177/commits/2be8970fb0bc68245a6f6b9d3a6f1c1270a3e4e6 and https://github.com/coreos/tectonic-installer/pull/2177/commits/44f1ce54b0bd1b275312e13aa6a76b947d291770 ?

**Pull request merged**